sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -298,16 +302,34 @@ def _error(outcome, condition):
         if outcome != constants.MessageSendResult.Ok:
             raise condition
 
+    def create_batch(self, max_size=None, partition_key=None):
+        """
+        Create an EventDataBatch object with max message size being max_message_size.
+        The max_message_size should be no greater than the max allowed message size defined by the service side.
+        :param max_message_size:
+        :param partition_key:
+        :return:
+        """
+        if not self._max_message_size_on_link:
+            self._open()
```

Comment:
If we change the behaviour for the async create_batch (to raise an error if the connection isn't open) we will need to do the same thing here.

Reply:
Same to the above async one. We should open internally first to get the right peer link message size.